src/file.rs

```diff
@@ -233,29 +251,34 @@ impl AsepriteFile {
         image
     }
 
+    fn resolve_indexed_pixel(
+        &self,
+        layer_is_background: bool,
+        pixel: &pixel::Indexed,
+    ) -> pixel::Rgba {
+        let palette = self.palette.as_ref().expect("Expected a palette present when resolving indexed image. Should have been caught in validation");
+        let transparent_color_index = self.pixel_format.transparent_color_index().expect("Indexed tilemap pixels in non-indexed pixel format. Should have been caught in validation");
+        pixel
```

Comment:
Since this function is called per-pixel it probably shouldn't have so many failure points. These first two should always be valid after the first invocation of the function. So, maybe we can make this a standalone function (i.e., no `self` argument) and pass in the palette reference and transparent color index as arguments. Then do the checks where you construct index resolver closure below.

Reply:
The closure would still need to be constructed conditionally to avoid unwrapping stuff when the closure's not needed. I briefly tried a "closure factory" signature (e.g `|| -> |px| -> rgba`) and that caused some lifetime issues. I decided to just have `Pixels` accept the optional arguments, and in the Indexed case, do the unwrapping, and then iterate.